src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,102 @@
+package duke;
+
+import java.time.format.DateTimeParseException;
+
+/**
+ * Parser Class is responsible for interpreting the commands entered by the user. Upon invalid commands, exceptions
+ * will be thrown and handled.
+ */
+public class Parser {
+    private String command = "";
+    private String secondaryInput = "";
+
+    /**
+     * Constructor for the Parser Class.
+     */
+    public Parser() {
+
+    }
+
+    /**
+     * Method is responsible for interpreting commands send by the user and handling invalid commands.
+     * @param commandInput
+     * @param duke
+     * @param taskList
+     */
+    public String input(String commandInput, Duke duke, TaskList taskList) {

```

Comment:
Since this is a method, following the coding convention, maybe you could give the method a verb name instead?
For example, public String parseInput(...) could work.